includes/scheduler/class-comment.php

```diff
@@ -84,4 +73,16 @@ public static function schedule_comment_activity( $new_status, $old_status, $com
 
 		add_to_outbox( $comment, $type, $comment->user_id );
 	}
+
+	/**
+	 * Schedule Comment Activities on insert.
+	 *
+	 * @param int         $comment_id Comment ID.
+	 * @param \WP_Comment $comment    Comment object.
+	 */
+	public static function schedule_comment_activity_on_insert( $comment_id, $comment ) {
+		if ( 1 === (int) $comment->comment_approved ) {
+			self::schedule_comment_activity( 'approved', '', $comment );
+		}
```

Comment:
Should a comment Create activity only be added when the comment was approved?

Reply:
Sure!

I think this was not checked yet, because we only federate comments of registered blog users and I think they will be auto-approved!?!

But better to have it checked explicitly 👍